plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -0,0 +1,38 @@
+{-# LANGUAGE ConstraintKinds  #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs            #-}
+{-# LANGUAGE TypeFamilies     #-}
+
+{-|
+Call site inlining machinery. For now there's only one part: inlining of fully applied functions.
+We inline fully applied functions if the cost and size are acceptable.
+-}
+
+module PlutusIR.Transform.Inline.CallSiteInline where
+
+import PlutusIR
+import PlutusIR.Transform.Rename ()
+import PlutusPrelude
+
+-- | Count the number of type and term lambdas in the RHS of a binding
+countLam ::
```

Comment:
Many of these functions are generic functions that don't necessarily uniquely relate to inlining, so if we're going to move them I think we might as well put them in some more generic utility module.

Reply:
I think that's "too small" IMO. It would be a bit different if it was something that was a bit bigger or of obviously utility that we're definitely going to use elsewhere, but this is so tiny and specific I think it's not worth it.